projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -35,19 +35,18 @@ import {
 import {TuiCountryIsoCode} from '@taiga-ui/i18n';
 import {TUI_ARROW} from '@taiga-ui/kit/components/arrow';
 import {TuiInputPhoneComponent} from '@taiga-ui/kit/components/input-phone';
+import {TuiExtractCountryCodePipe} from '@taiga-ui/kit/pipes';
 import {FIXED_DROPDOWN_CONTROLLER_PROVIDER} from '@taiga-ui/kit/providers';
-import {TUI_COUNTRIES} from '@taiga-ui/kit/tokens';
+import {TUI_COUNTRIES, TUI_COUNTRIES_MASKS} from '@taiga-ui/kit/tokens';
+import {tuiGetMaxAllowedPhoneLength, tuiIsoToCountryCode} from '@taiga-ui/kit/utils';
```

Comment:
it would be great if you introduce

```ts
export interface TuiInternationalPhoneOptions {
    readonly allowedPhoneLength: TuiHandler<?, ?>;
    readonly isoToCountryCode: TuiHandler<?, ?>;
}

export const TUI_INTERNATIONAL_DEFAULT_OPTIONS: TuiInternationalPhoneOptions = {
    allowedPhoneLength: tuiGetMaxAllowedPhoneLength,
    isoToCountryCode: tuiIsoToCountryCode
};

export const TUI_INTERNATIONAL_OPTIONS = new InjectionToken<TuiInternationalPhoneOptions>(
    `[TUI_RATING_OPTIONS]`,
    {
        factory: () => TUI_INTERNATIONAL_DEFAULT_OPTIONS,
    },
);
```

and then

```ts
   get inputPhoneCountryCode(): string {
        return this.options.isoToCountryCode(this.countriesMasks, this.countryIsoCode);
    }

    get phoneMaskAfterCountryCode(): string {
        const countryCode = this.options.isoToCountryCode(this.countriesMasks, this.countryIsoCode);
    // ....


   // ...
   .slice(
                    0,
                    this.options.allowedPhoneLength(this.countriesMasks, this.countryIsoCode),
                );


```

Reply:
TuiHandler<?, ?> can have only 1 argument, but both functions require 2 arguments.